pkg/services/datasources/service/store.go

```diff
@@ -253,7 +253,7 @@ func (ss *SqlStore) AddDataSource(ctx context.Context, cmd *datasources.AddDataS
 				return fmt.Errorf("failed to generate UID for datasource %q: %w", cmd.Name, err)
 			}
 			cmd.UID = uid
-		} else if !util.IsValidShortUID(cmd.UID) {
+		} else if err := util.ValidateUID(cmd.UID); err != nil {
```

Comment:
Maybe we should include the error in the log as well?

Reply:
Good point, added